Remove ldap_str2dn compatibility kludge

This avoids calling the internal, undeclared ldap_bv2dn_x function.
This might be an actual bug because the internal function has since
added an additional argument (which would have to be specified as NULL
in this case).

Submitted upstream:

  <http://lists.askja.de/pipermail/ldapvi/2022-December/000127.html>

diff --git a/diff.c b/diff.c
index 0faaf8b790ab740b..3610563450e0b694 100644
--- a/diff.c
+++ b/diff.c
@@ -201,26 +201,7 @@ frob_ava(tentry *entry, int mode, char *ad, char *data, int n)
 	return 0;
 }
 
-#if defined(LIBLDAP21)
-#warning compiling for libldap <= 2.1, running with >= 2.2 will result in segfault
 #define safe_str2dn ldap_str2dn
-#elif defined(LIBLDAP22)
-/*
- * the following is exactly equivalent to ldap_str2dn in libldap >= 2.2,
- * but will fail linking on 2.1.  This way we avoid calling the old 2.1
- * version of ldap_str2dn (leading to a segfault when accessing the result).
- */
-static void
-safe_str2dn(char *str, LDAPDN *out, int flags)
-{
-        struct berval bv;
-        bv.bv_val = str;
-        bv.bv_len = strlen(str);
-        ldap_bv2dn_x(&bv, out, flags);
-}
-#else
-#error oops
-#endif
 
 /*
  * Call frob_ava for every ava in DN's (first) RDN.
